crates/cli/src/parse/timestamps.rs

```diff
@@ -311,44 +312,40 @@ async fn timestamp_to_block_number(timestamp: u64, source: Arc<Source>) -> Resul
 async fn get_latest_timestamp(source: Arc<Source>) -> Result<u64, ParseError> {
     let latest_block_number = get_latest_block_number(source.clone()).await?;
     let latest_block = source
-        .get_block(latest_block_number)
+        .get_block(latest_block_number, BlockTransactionsKind::Hashes)
         .await
         .map_err(|_e| ParseError::ParseError("Error fetching latest block".to_string()))?
         .unwrap();
 
-    Ok(latest_block.timestamp.as_u64())
+    Ok(latest_block.header.timestamp)
 }
 
 #[cfg(test)]
 mod tests {
     use std::num::NonZeroU32;
 
+    use alloy::providers::ProviderBuilder;
     use governor::{Quota, RateLimiter};
 
     use super::*;
     use cryo_freeze::SourceLabels;
-    use ethers::prelude::*;
 
     async fn setup_source() -> Source {
         let rpc_url = match crate::parse::source::parse_rpc_url(&Args::default()) {
             Ok(url) => url,
             Err(_) => std::process::exit(0),
         };
-        let max_retry = 5;
-        let initial_backoff = 500;
+        // let max_retry = 5;
+        // let initial_backoff = 500;
         let max_concurrent_requests = 100;
-        let provider =
-            Provider::<RetryClient<Http>>::new_client(&rpc_url, max_retry, initial_backoff)
-                .map_err(|_e| ParseError::ParseError("could not connect to provider".to_string()))
-                .unwrap();
-
+        let provider = ProviderBuilder::new().on_http(rpc_url.parse().unwrap()).boxed();
```

Comment:
need to set client with a RetryBackupLayer as in https://alloy.rs/examples/layers/retry_layer.html